src/components/AddItemForm.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { withFirestore } from 'react-firestore';
+import '../CSS/AddItemForm.css';
+class AddItemForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      next_purchase: null,
+      last_purchase: null,
+      user_token: 'shpongle',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
```

Comment:
Not a blocker, just FYI in case you didn't know. With create-react-apps (CRA), we have a babel feature that will let you define method without [explicitly binding](https://reactjs.org/docs/react-without-es6.html#autobinding) `this` in the constructor.

```jsx
handleChange = event => {
  this.setState({ name: event.target.value });
}
```

Like the docs mention, it's an experimental feature so your mileage may vary. But it's definitely a common pattern.